src/main/java/jacchm/footballapp/service/impl/FootballDataOrgServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package jacchm.footballapp.service.impl;
+
+import jacchm.footballapp.customexceptions.ExternalFootballApiConnectionException;
+import jacchm.footballapp.service.FootballDataOrgService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.text.MessageFormat;
+import java.util.Date;
+
+@Slf4j
+@Service
+public class FootballDataOrgServiceImpl implements FootballDataOrgService {
+
+    private final String FOOTBALL_API_URL = "http://api.football-data.org";
+    private final HttpMethod HTTP_METHOD = HttpMethod.GET;
+    private final String AUTH_TOKEN;
+    private final String AUTH_HEADER = "X-Auth-Token";
+
+    public FootballDataOrgServiceImpl(@Value("${AUTH_TOKEN}") String auth_token) {
+        AUTH_TOKEN = auth_token;
+    }
+
+    @Override
+    public String getCompetitions() throws ExternalFootballApiConnectionException {
+        String url = FOOTBALL_API_URL + "/v2/competitions/";
+
+        return getResponse(getRequestEntity(url));
+    }
+
+    @Override
+    public String getTeams(int competitionId) throws ExternalFootballApiConnectionException {
+        String url = MessageFormat.format(FOOTBALL_API_URL + "/v2/competitions/{0}/teams",
+                Integer.toString(competitionId));
+
+        return getResponse(getRequestEntity(url));
+    }
+
+    @Override
+    public String getResults(int competitionId) throws ExternalFootballApiConnectionException {
+        String url = MessageFormat.format(FOOTBALL_API_URL + "/v2/competitions/{0}/standings",
+                Integer.toString(competitionId));
+
+        return getResponse(getRequestEntity(url));
+    }
+
+    private String getResponse(RequestEntity request) throws ExternalFootballApiConnectionException {
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(request, String.class);

```

Comment:
Jak zrobisz dedykowane getForObject zamiast exchange to od razu zmapuje Ci Spring na obiekt i nie będziesz się musiał bawić potem mapperem - JsonUtil w ogóle do wywalenia wtedy.